openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -655,15 +656,8 @@ def _validate_password(password, username=None):
     try:
         _validate_type(password, basestring, accounts.PASSWORD_BAD_TYPE_MSG)
 
-        if len(password) == 0:
-            raise errors.AccountPasswordInvalid(accounts.PASSWORD_EMPTY_MSG)
-        elif len(password) < accounts.PASSWORD_MIN_LENGTH:
-            raise errors.AccountPasswordInvalid(accounts.PASSWORD_BAD_MIN_LENGTH_MSG)
-        elif len(password) > accounts.PASSWORD_MAX_LENGTH:
-            raise errors.AccountPasswordInvalid(accounts.PASSWORD_BAD_MAX_LENGTH_MSG)
-
-        _validate_password_works_with_username(password, username)
-    except (errors.AccountDataBadType, errors.AccountDataBadLength) as invalid_password_err:
+        validate_password(password, username=username)
```

Comment:
This here is the original point of this PR (to check fuller policy settings when live validating on the combined login/registration form). The rest of this branch is refactoring to consolidate code paths.